marshmallow/fields.py

```diff
@@ -445,35 +439,50 @@ def _serialize(self, nested_obj, attr, obj):
             schema._update_fields(obj=nested_obj, many=self.many)
             self.__updated_fields = True
         try:
-            ret = schema.dump(nested_obj, many=self.many,
-                              update_fields=not self.__updated_fields)
+            return schema.dump(nested_obj, many=self.many,
+                               update_fields=not self.__updated_fields)
         except ValidationError as exc:
             raise ValidationError(exc.messages, data=obj, valid_data=exc.valid_data)
-        finally:
-            if isinstance(self.only, basestring):  # self.only is a field name
-                only_field = self.schema.fields[self.only]
-                key = ''.join([self.schema.prefix or '', only_field.data_key or self.only])
-                if self.many:
-                    return utils.pluck(ret, key=key)
-                else:
-                    return ret[key]
-        return ret
 
-    def _deserialize(self, value, attr, data):
+    def test_collection(self, value):
```

Comment:
Why not make this method private `_test_collection`?
Same with `load` -> `_load`.

Reply:
OK, I just realized you're using them in `Pluck`. Maybe that's the reason you're exposing them as public API. I don't think using them in a child class here means they need to be exposed publicly (or does it?), but maybe your point is to expose them for other potential child classes to use?

Reply:
In C++ it would be considered a "protected" member, but it was really just needed to decompose `_deserialize` into reusable pieces. Prefixing it with `_` is probably fine since it doesn't need to be used outside subclasses.